docs/rst/container_yml/reference.rst

```diff
@@ -758,3 +844,67 @@ requested_storage        The amount of storage being requested. Defaults to 1Gi.
 ======================== =============================================================================================================
 
 
+.. secrets:
+
+Secrets
+```````
+
+For Docker, the service level ``secrets`` directive works as expected. The top-level ``secrets`` directive, however, has been modified slightly. The following example ``container.yml`` shows the three forms of the service level ``secrets`` directive, and the new top-level ``secrets`` format:
+
+.. code-block:: yaml
+
+    version: '2'
+    services:
+      web:
+        from: centos:7
+        entrypoint: [/usr/bin/entrypoint.sh]
+        working_dir: /
+        user: apache
+        command: [/usr/bin/dumb-init, httpd, -DFOREGROUND]
+        ports:
+        - 8000:8080
+        - 4443:8443
+        roles:
+        - apache-container
+        secrets:
+          - apache-certs
+
+    secrets:
+      my-secrets:
+        docker: {}
+        openshift:
+          state: present
+          force: false
+          type: Generic
+          data:
+            username: admin
+            password:
+              type: literal
```

Comment:
This could be switched possibly to `from` to reflect the CLI usage: `create secret generic --from-literal`